Build failure with mtl-2.3

```
Building library for units-parser-0.1.1.4..
[1 of 1] Compiling Text.Parse.Units ( Text/Parse/Units.hs, dist/build/Text/Parse/Units.o, dist/build/Text/Parse/Units.dyn_o )

Text/Parse/Units.hs:94:10: error:
    Variable not in scope:
      void :: ParsecT s0 u1 m0 Char -> ParsecT s u m ()
   |
94 | nochar = void . char
   |          ^^^^

Text/Parse/Units.hs:271:24: error:
    • Variable not in scope:
        liftM
          :: ([Maybe a2] -> [a2])
             -> ParsecT
                  String () (Reader (SymbolTable pre u3)) [Maybe (pre, u3)]
             -> ParsecT String () (Reader (SymbolTable pre u)) [(pre, u)]
    • Perhaps you meant ‘lift’ (imported from Control.Monad.Reader)
    |
271 |   results <- catMaybes `liftM` mapM (experiment . parse_one) assocs
    |                        ^^^^^^^

Text/Parse/Units.hs:284:7: error:
    Variable not in scope:
      void
        :: ParsecT s1 u2 m1 String
           -> ParsecT String () (Reader (SymbolTable pre5 u8)) a1
    |
284 |       void $ string pre
    |       ^^^^

Text/Parse/Units.hs:290:38: error:
    • Variable not in scope:
        liftM
          :: (u4 -> UnitExp pre1 u4)
             -> GenUnitStringParser pre2 u5 u5
             -> ParsecT String () (Reader (SymbolTable pre u)) (UnitExp pre u)
    • Perhaps you meant ‘lift’ (imported from Control.Monad.Reader)
    |
290 | unitStringParser = try (Unit Nothing `liftM` justUnitP) <|> prefixUnitP
    |                                      ^^^^^^^

Text/Parse/Units.hs:342:13: error:
    • Variable not in scope:
        liftM
          :: (a0 -> a0)
             -> GenUnitParser pre0 u0 Integer
             -> ParsecT [Token] () (Reader (SymbolTable pre u)) Integer
    • Perhaps you meant ‘lift’ (imported from Control.Monad.Reader)
    |
342 |      negate `liftM` numP
    |             ^^^^^^^

Text/Parse/Units.hs:381:16: error:
    • Variable not in scope:
        liftM
          :: (t0 (UnitExp pre3 u6) -> UnitExp pre3 u6)
             -> ParsecT
                  [Token] () (Reader (SymbolTable pre4 u7)) [UnitExp pre4 u7]
             -> ParsecT [Token] () (Reader (SymbolTable pre u)) (UnitExp pre u)
    • Perhaps you meant ‘lift’ (imported from Control.Monad.Reader)
    |
381 |   (foldl1 Mult `liftM` many1 unitP)
    |                ^^^^^^^
```

--- a/Tests/Parser.hs
+++ b/Tests/Parser.hs
@@ -10,7 +10,8 @@ import Prelude hiding ( lex, exp )
 
 import Text.Parse.Units
 
-import Control.Monad.Reader
+import Control.Monad (liftM)
+import Control.Monad.Reader (runReader)
 import qualified Data.Map.Strict as Map
 import Text.Parsec
 import Data.Generics

--- a/Text/Parse/Units.hs
+++ b/Text/Parse/Units.hs
@@ -52,7 +52,8 @@ import Text.Parsec.String
 import Text.Parsec.Pos
 import qualified Data.Map.Strict as Map
 import qualified Data.MultiMap as MM
-import Control.Monad.Reader
+import Control.Monad (liftM, void)
+import Control.Monad.Reader (MonadReader(..), Reader, asks, runReader)
 import Control.Arrow       hiding ( app)
 import Data.Data (Data)
 import Data.Maybe

